File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/3656-minimum-number-of-operations-to-make-elements-in-array-distinct.py
class Solution:
    def minimumOperations (self, nums: List [int]) -> int:
        ans = 0
        while True:
            count = defaultdict (int)
            isDistinct = True
            for n in nums [ans * 3 :]:
                count [n] += 1
                if count [n] > 1:
                    isDistinct = False
                    break
            if isDistinct:
                return ans
            ans += 1

File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/3656-minimum-number-of-operations-to-make-elements-in-array-distinct.c
#define CONSTRAINT 101

int minimumOperations (int* nums, int numsSize) {
    int freq [CONSTRAINT] = {0};
    for (int i = numsSize - 1; i >= 0; freq [nums [i--]] = 1)
        if (freq [nums [i]]) return (i + 1) / 3 + ((i + 1) % 3 ? 1 : 0);
    return 0;
}